src/test/java/org/springframework/data/redis/core/DefaultStreamOperationsIntegrationTests.java

```diff
@@ -206,6 +206,19 @@ void addMaxLenShouldLimitSimpleMessagesSize() {
 		assertThat(message.getValue()).isEqualTo(newValue);
 	}
 
+	@ParameterizedRedisTest // GH-2982
+	void addNegativeMaxlenShouldThrowException() {
+
+		K key = keyFactory.instance();
+		HV value = hashValueFactory.instance();
+
+		XAddOptions options = XAddOptions.maxlen(-1).approximateTrimming(false);
+
+		assertThatThrownBy(() -> streamOps.add(StreamRecords.objectBacked(value).withStreamKey(key), options));
```

Comment:
Testing Redis/Driver behavior exceeds the scope of what we want to test. We will remove these tests during the merge.